google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1alpha2/JsonToProtoMessageTest.java

```diff
@@ -0,0 +1,482 @@
+    JSONObject json = new JSONObject();
+    json.put("test_repeated", new JSONArray("[1.1, 2.2, true]"));
+    try {
+      DynamicMessage protoMsg =
+          JsonToProtoMessage.convertJsonToProtoMessage(RepeatedDouble.getDescriptor(), json);
+    } catch (IllegalArgumentException e) {
+      assertEquals(
+          e.getMessage(), "JSONObject does not have a double field at root.test_repeated[2].");
+    }
+  }
+
+  @Test
+  public void testNestedRepeatedComplex() throws Exception {
+    double[] doubleArr = {1.1, 2.2, 3.3, 4.4, 5.5};
+    String[] stringArr = {"hello", "this", "is", "a", "test"};
+    int[] intArr = {1, 2, 3, 4, 5};
+
+    JSONObject json = new JSONObject();
+    json.put("double", new JSONArray(doubleArr));
+    json.put("int", new JSONArray(intArr));
+    JSONObject jsonRepeatedString = new JSONObject();
+    jsonRepeatedString.put("test_repeated", new JSONArray(stringArr));
+    json.put("repeated_string", jsonRepeatedString);
+
+    DynamicMessage protoMsg =
+        JsonToProtoMessage.convertJsonToProtoMessage(NestedRepeated.getDescriptor(), json);
+    AreMatchingFieldsFilledIn(protoMsg, json);
+  }
+
+  @Test
+  public void testNestedRepeatedComplexFail() throws Exception {
+    double[] doubleArr = {1.1, 2.2, 3.3, 4.4, 5.5};
+    Boolean[] fakeStringArr = {true, false};
+    int[] intArr = {1, 2, 3, 4, 5};
+
+    JSONObject json = new JSONObject();
+    json.put("double", new JSONArray(doubleArr));
+    json.put("int", new JSONArray(intArr));
+    JSONObject jsonRepeatedString = new JSONObject();
+    jsonRepeatedString.put("test_repeated", new JSONArray(fakeStringArr));
+    json.put("repeated_string", jsonRepeatedString);
+
+    try {
+      DynamicMessage protoMsg =
+          JsonToProtoMessage.convertJsonToProtoMessage(NestedRepeated.getDescriptor(), json);
+    } catch (IllegalArgumentException e) {
+      assertEquals(
+          e.getMessage(),
+          "JSONObject does not have a string field at root.repeated_string.test_repeated[0].");
+    }
+  }
+
+  @Test
+  public void testAllowUnknownFields() throws Exception {
+    JSONObject json = new JSONObject();
+    json.put("test_repeated", new JSONArray(new int[] {1, 2, 3, 4, 5}));
+    json.put("string", "hello");
+
+    DynamicMessage protoMsg =
+        JsonToProtoMessage.convertJsonToProtoMessage(RepeatedInt64.getDescriptor(), json, true);
+    AreMatchingFieldsFilledIn(protoMsg, json);
+  }
+
+  @Test
+  public void testAllowUnknownFieldsError() throws Exception {
+    JSONObject json = new JSONObject();
+    json.put("double", 1.1);
+    json.put("string", "hello");
+
+    try {
+      DynamicMessage protoMsg =
+          JsonToProtoMessage.convertJsonToProtoMessage(RepeatedInt64.getDescriptor(), json);
+    } catch (IllegalArgumentException e) {
+      assertEquals(
+          e.getMessage(),
+          "JSONObject has unknown fields. Set allowUnknownFields to True to ignore unknown fields.");
```

Comment:
Let's put the error message like this: JSONObject has fields unknown to BigQuery: f1. Set allowUnknownFields to True to allow unknown fields.

Reply:
Got it

Reply:
Does the f1 mean anything?

Reply:
No it is just an example.